tests/PHPStan/Rules/Variables/data/bug-10610.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace Bug10610;
+
+use function PHPStan\Testing\assertType;
+
+class HelloWorld
+{
+	private const MAP = [
+		'foo' => [
+			'19' => '582',
+			'26' => '689',
+			'56' => '817',
+			'52' => '1050',
+			'67' => '2923',
+			'78' => '4057',
+			'75' => '4078',
+			'54' => '4078',
+			'76' => '4079',
+			'77' => '4080',
+			'9' => '4080',
+			'46' => '4091',
+			'22' => '4111',
+			'48' => '4112',
+			'70' => '4113',
+			'42' => '4117',
+			'43' => '4118',
+			'6' => '4126',
+			'36' => '4129',
+			'13' => '4309',
+			'14' => '4904',
+			'5' => '5222',
+			'71' => '5223',
+			'73' => '5242',
+			'74' => '5250',
+			'24' => '5252',
+			'58' => '5255',
+			'35' => '5261',
+			'1' => '5264',
+			'20' => '5268',
+			'21' => '5269',
+			'31' => '5270',
+			'51' => '5271',
+			'55' => '5271',
+			'39' => '5274',
+			'50' => '5277',
+			'49' => '5278',
+			'11' => '5279',
+			'41' => '5279',
+			'44' => '5280',
+			'59' => '5281',
+			'60' => '5281',
+			'23' => '5281',
+			'72' => '5283',
+			'32' => '5283',
+			'8' => '5285',
+			'40' => '5285',
+			'12' => '5298',
+			'37' => '5305',
+			'65' => '5310',
+			'64' => '5310',
+			'57' => '5352',
+			'33' => '5364',
+			'25' => '5375',
+			'34' => '5460',
+			'45' => '7581',
+			'3' => '7624',
+			'53' => '7672',
+			'999' => '7953',
+			'69' => '7953',
+			'2' => '8206',
+			'7' => '9697',
+		],
+		'bar' => [
+			'30' => 'Test3',
+		],
+	];
+
+	public function validate(string $k, string $value): void
+	{
+		$res = self::MAP[$k][$value] ?? '';
+
+		assertType("'582'|'689'|'817'|'1050'|'2923'|'4057'|'4078'|'4078'|'4079'|'4080'|'4080'|'4091'|'4111'|'4112'|'4113'|'4117'|'4118'|'4126'|'4129'|'4309'|'4904'|'5222'|'5223'|'5242'|'5250'|'5252'|'5255'|'5261'|'5264'|'5268'|'5269'|'5270'|'5271'|'5271'|'5274'|'5277'|'5278'|'5279'|'5279'|'5280'|'5281'|'5281'|'5281'|'5283'|'5283'|'5285'|'5285'|'5298'|'5305'|'5310'|'5310'|'5352'|'5364'|'5375'|'5460'|'7581'|'7624'|'7672'|'7953'|'7953'|'8206'|'9697'|'Test3'", self::MAP[$k][$value]);
```

Comment:
You need to do this in a file that's referenced from NodeScopeResolverTest

Reply:
Never trust a test you haven't seen fail 😊

Reply:
I am very lost here, I have added this: https://github.com/phpstan/phpstan-src/pull/2928/files#diff-1e9b47d3e7051caa11f9d76ff7bd6ff2afb430014dd93dad916cbb8c2148f545R171

Sorry 🤦‍♂️

Reply:
The CI results show that the `getOffsetValueType` needs updating.

Reply:
Also feel free to make the array shorter, we don't need such a big array to verify this functionality. Thanks.

Reply:
Sorry @ondrejmirtes , i do not follow, the error message is:
```diff
--- Expected
+++ Actual
@@ @@
-''582'|'689'|'817'|'1050'|'2923'|'4057'|'4078'|'4078'|'4079'|'4080'|'4080'|'4091'|'4111'|'4112'|'4113'|'4117'|'4118'|'4126'|'4129'|'4309'|'4904'|'5222'|'5223'|'5242'|'5250'|'5252'|'5255'|'5261'|'5264'|'5268'|'5269'|'5270'|'5271'|'5271'|'5274'|'5277'|'5278'|'5279'|'5279'|'5280'|'5281'|'5281'|'5281'|'5283'|'5283'|'5285'|'5285'|'5298'|'5305'|'5310'|'5310'|'5352'|'5364'|'5375'|'5460'|'7581'|'7624'|'7672'|'7953'|'7953'|'8206'|'9697'|'Test3''
+'*ERROR*'
```

I bet you do not want me replace all these types with `'*ERROR*'` ❓

Reply:
Updated also types, just unique values.